test/jdk/javax/swing/JProgressBar/TestProgressBarBorder.java

```diff
@@ -0,0 +1,160 @@
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    pt = progressBar.getLocationOnScreen();
+                });
+
+                BufferedImage borderPaintedImg =
+                        robot.createScreenCapture(new Rectangle(pt.x, pt.y,
+                                progressBar.getWidth(), progressBar.getHeight()));
+
+                progressBar.setBorderPainted(false);
+                robot.waitForIdle();
+                robot.delay(500);
+
+                BufferedImage borderNotPaintedImg =
+                        robot.createScreenCapture(new Rectangle(pt.x, pt.y,
+                                progressBar.getWidth(), progressBar.getHeight()));
+
+                robot.delay(500);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    passed = compareImage(borderPaintedImg, borderNotPaintedImg);
+                });
+
+                if (!passed) {
+                    ImageIO.write(borderPaintedImg, "png", new File("borderPaintedImg.png"));
+                    ImageIO.write(borderNotPaintedImg, "png", new File("borderNotPaintedImg.png"));
+                    throw new RuntimeException("JProgressBar border is painted although " +
+                            "border painting is set to false");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void createAndShowUI() {
+        frame = new JFrame("Test JProgressBar Border");
+        JPanel p = new JPanel(new FlowLayout());
+        progressBar  = new JProgressBar();
+        // set initial value
+        progressBar.setValue(0);
+        progressBar.setBorderPainted(true);
+        progressBar.setStringPainted(true);
+        p.add(progressBar);
+        frame.add(p);
+        frame.setSize(200, 100);
+        frame.setLocationRelativeTo(null);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setVisible(true);
+    }
+
+    /*
+    * Compare JProgressBar border painted and border not painted image and
+    * if both images width and height are equal but pixel's RGB values are
+    * not equal, method returns true; false otherwise.
+    */
+
+    private static boolean compareImage(BufferedImage img1, BufferedImage img2) {
```

Comment:
You can use the method from `regtesthelpers/Util`:

https://github.com/openjdk/jdk/blob/9faead1469481e268b451f2853c8fec8613426b9/test/jdk/javax/swing/regtesthelpers/Util.java#L59-L80

If you don't want to use, I suggest reversing the conditions so that `compareImage` returns `true` when images are the same — it's more common and therefore less confusing. In addition to that, return quickly if the sizes are different (can they ever be?) and avoid nesting the `for`-loops inside the `if` statement.

Reply:
Updated the test to use the method from regtesthelpers/Util.